### PR TITLE
Data tab uitest

### PR DESCRIPTION
**Pull request by maddiedierker**
Basic UI test coverage for the data tab in Applab: importing a dataset, creating/editing tables/rows, and creating/editing key/value pairs.

## Links

- [STAR-1148](https://codedotorg.atlassian.net/browse/STAR-1148)

## Testing story

Adds UI test coverage.

# Reviewer Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: dashboard/test/ui/features/star_labs/applab/data_tab.feature [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    And I click selector ".dropdown.pull-right:first-of-type"
+    And I press keys "number column" for element ".uitest-data-table-content:first-of-type th:nth-child(2) input"
+    When I press enter key
+    Then I wait until element ".test-tableNameDiv:first-of-type" contains text "number column"
+
+    # Add a row
+    And I click selector "#addDataTableRow input:first-of-type"
+    And I press keys "2" for element "#addDataTableRow input:first-of-type"
+    And I click selector "#addDataTableRow button:first-of-type"
+    Then I wait until element ".uitest-data-table-content:first-of-type tr:nth-child(3) td:nth-child(2)" contains text "2"
```
Comment (breville) [inline dashboard/test/ui/features/star_labs/applab/data_tab.feature]: Is it worth adding a unique identifier for some of these more complex selectors?  Not sure whether this selector might be a bit fragile long term...

Comment (breville) [reply dashboard/test/ui/features/star_labs/applab/data_tab.feature]: Alternatively, maybe these table-specific lines could be factored out into a helper?

Comment (maddiedierker) [reply dashboard/test/ui/features/star_labs/applab/data_tab.feature]: yeah, especially coming back to this PR after putting it down for a while, these are difficult to read. i'll start with adding classes to the row (and other) elements that have lengthy selectors

**Comment by maddiedierker** [comment]
@breville PTAL! i simplified some of the selectors and suresh made the `FIREBASE_SHARED_SECRET` env variable available on drone, so this test is in a good state now

**Review by breville — APPROVED** [review]
Nice.  These shorter selectors are much more readable!  

If we referenced these tables a ton we could probably write some more helpers to generate the selectors, but given the size of the actual test here, this seems like a good balance.

**Pull request merged**